content/spin/rust-components.md

```diff
@@ -367,6 +367,61 @@ messages on the `messages` Redis channel.
 
 Spin has a key-value store built in. For information about using it from Rust, see [the key-value store tutorial](kv-store).
 
+If you want to go beyond just inserting values in a certain spot, Spin's Rust SDK provides an optional feature for using [Serde](https://docs.rs/serde/latest/serde). The Rust code below shows how to store and retrieve serializable objects from the key-value store:
+
+```rust
+use anyhow::Result;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+    key_value::{Store},
+};
+use serde::{Serialize, Deserialize};
+
+#[http_component]
+fn handle_request(_req: Request) -> Result<Response> {
+    // Open the default key-value store
+    let store = Store::open_default()?;
+    // Define a User object using serde derive
+    #[derive(Serialize, Deserialize, Debug)]
+    struct User {
+        fingerprint: String,
+        location: String,
+    }
+    // Create an instance of a user object and populate the values
+    let user = User{fingerprint:"0x1234".to_owned(), location:"Brisbane".to_owned()};
```

Comment:
nit: can we format these snippets with rustfmt?

Reply:
Sure think; sorry, I should have done that in the first place.
Formatting will be in the next commit.